Guide client integration with torii

respp opened the pull request:
# Documentation Restructure Summary

## 🎯 Problem Solved

**Original Issue**: The Torii client integration guide was too long (1,176 lines) and difficult to navigate.

**Solution**: Broke down the large documentation into smaller, focused guides that are easier to digest and navigate.

## 📊 Before vs After

### Before (Single Large File)
- **File**: `client-new/pages/deployment/torii/client-integration.md`
- **Size**: 1,176 lines
- **Issues**: 
  - Too long to read in one sitting
  - Difficult to find specific information
  - Overwhelming for new developers
  - Poor navigation experience

### After (Modular Structure)
- **Main Overview**: 168 lines (concise introduction)
- **6 Focused Guides**: 2,733 lines total
- **Benefits**:
  - Easy to navigate and find specific topics
  - Each guide focuses on one aspect
  - Better developer experience
  - Progressive learning path

## 📁 New Structure

```
client-new/pages/deployment/torii/
├── client-integration.md          # Main overview (168 lines)
└── guides/
    ├── setup.md                   # Setup & configuration (202 lines)
    ├── graphql-queries.md         # Query patterns (389 lines)
    ├── custom-hooks.md            # Hook implementation (510 lines)
    ├── data-conversion.md         # Data utilities (388 lines)
    ├── error-handling.md          # Error management (576 lines)
    └── performance.md             # Optimization (668 lines)
```

## 🎯 Guide Breakdown

### 1. **Main Overview** (`client-integration.md`)
- Quick start guide
- Basic setup instructions
- Links to detailed guides
- Working examples for immediate use

### 2. **Setup & Configuration** (`guides/setup.md`)
- Prerequisites and dependencies
- Environment configuration
- Dojo configuration setup
- Common setup issues and solutions

### 3. **GraphQL Queries** (`guides/graphql-queries.md`)
- Query structure and patterns
- Cairo model query examples
- Advanced query patterns
- Best practices and debugging

### 4. **Custom Hooks** (`guides/custom-hooks.md`)
- Basic hook structure
- Multiple model hooks
- Enhanced hooks with data conversion
- Auto-refetch patterns
- Component usage examples

### 5. **Data Conversion** (`guides/data-conversion.md`)
- Hex to number conversion
- Address formatting utilities
- Cairo enum handling
- Bulk data conversion
- Type-safe conversion patterns

### 6. **Error Handling** (`guides/error-handling.md`)
- Custom error classes
- Retry strategies
- Error boundaries
- Error monitoring
- Recovery strategies

### 7. **Performance** (`guides/performance.md`)
- Caching strategies
- Request batching
- Performance monitoring
- Memory management
- Query optimization

## 🧭 Navigation Improvements

### Updated Routes Structure
```typescript
{
  text: 'Torii Integration',
  items: [
    {
      text: 'Client Integration',
      link: '/deployment/torii/client-integration',
    },
    {
      text: 'Guides',
      items: [
        { text: 'Setup & Configuration', link: '/deployment/torii/guides/setup' },
        { text: 'GraphQL Queries', link: '/deployment/torii/guides/graphql-queries' },
        { text: 'Custom Hooks', link: '/deployment/torii/guides/custom-hooks' },
        { text: 'Data Conversion', link: '/deployment/torii/guides/data-conversion' },
        { text: 'Error Handling', link: '/deployment/torii/guides/error-handling' },
        { text: 'Performance', link: '/deployment/torii/guides/performance' },
      ],
    },
  ],
}
```

### Cross-References
- Each guide links to the next logical guide
- Back navigation to main overview
- Related documentation references
- Clear learning path progression

## 🎯 Developer Experience Improvements

### ✅ **Easier Navigation**
- Find specific topics quickly
- Progressive learning path
- Clear guide organization

### ✅ **Better Readability**
- Focused content per guide
- Appropriate length for each topic
- Clear section organization

### ✅ **Improved Maintainability**
- Easier to update specific topics
- Better version control
- Modular content structure

### ✅ **Enhanced Learning**
- Step-by-step progression
- Copy-paste ready examples
- Comprehensive coverage

## 📈 Content Quality

### **Total Lines**: 2,901 (vs original 1,176)
- **More comprehensive coverage**
- **Better organized information**
- **Additional examples and patterns**
- **Production-ready code**

### **Coverage Areas**
- ✅ Basic setup and configuration
- ✅ GraphQL query patterns
- ✅ Custom hook implementation
- ✅ Data type conversion
- ✅ Error handling strategies
- ✅ Performance optimization
- ✅ Security best practices
- ✅ Testing strategies

## 🚀 Benefits Achieved

### **For New Developers**
- Start with quick overview
- Follow progressive learning path
- Find specific information easily
- Copy-paste working examples

### **For Experienced Developers**
- Jump to specific topics
- Advanced patterns and optimizations
- Production-ready implementations
- Best practices and security

### **For Maintainers**
- Easier to update specific topics
- Better organization
- Modular structure
- Clear navigation

## 🎯 Success Metrics

### **Usability**
- ✅ Reduced cognitive load
- ✅ Better information architecture
- ✅ Improved navigation experience
- ✅ Progressive learning path

### **Content Quality**
- ✅ More comprehensive coverage
- ✅ Better organized information
- ✅ Production-ready examples
- ✅ Best practices included

### **Maintainability**
- ✅ Easier to update
- ✅ Modular structure
- ✅ Clear separation of concerns
- ✅ Better version control

respp committed:
docs: add Torii client integration overview guide

- Create main overview page with quick start guide
- Include basic setup instructions and working examples
- Add links to detailed guides for comprehensive learning
- Provide copy-paste ready code for immediate use
- Address GitHub issue #194 for Torii client integration documentation

respp committed:
docs: add Torii setup and configuration guide

- Add comprehensive setup instructions for Torii integration
- Include environment configuration for different networks
- Provide Dojo configuration examples with security best practices
- Add troubleshooting section for common setup issues
- Include address validation and security considerations

respp committed:
docs: add GraphQL queries guide for Torii integration

- Add comprehensive GraphQL query structure documentation
- Include Cairo model query examples (Position, Moves)
- Provide advanced query patterns (pagination, filtering, sorting)
- Add query optimization best practices
- Include debugging and schema introspection examples

respp committed:
docs: add custom hooks guide for Torii integration

- Add comprehensive React hooks patterns for Torii data fetching
- Include basic, enhanced, and auto-refetch hook examples
- Provide multiple model query patterns and list hooks
- Add component usage examples with error handling
- Include hook best practices and performance considerations

respp committed:
docs: add data conversion utilities guide for Torii integration

- Add hex to number conversion utilities for Cairo model data
- Include address formatting and validation utilities
- Provide Cairo enum handling patterns (Direction enum examples)
- Add bulk data conversion and type-safe conversion patterns
- Include error handling and validation utilities for data conversion

respp committed:
docs: add error handling strategies guide for Torii integration

- Add comprehensive error handling patterns for Torii queries
- Include custom error classes (ToriiError, GraphQLError, NetworkError)
- Provide retry strategies with exponential backoff
- Add error boundaries and component error handling
- Include error monitoring, logging, and recovery strategies

respp committed:
docs: add performance optimization guide for Torii integration

- Add caching strategies with TTL and LRU implementations
- Include request batching and query deduplication patterns
- Provide performance monitoring and metrics collection
- Add memory management and query optimization techniques
- Include performance best practices and optimization guidelines

respp committed:
feat: add navigation for Torii integration guides

- Add Torii Integration section to deployment navigation
- Include links to all 6 detailed guides (setup, queries, hooks, etc.)
- Add cross-reference in React Integration section
- Update routes structure for proper navigation integration
- Ensure all guides are discoverable in the documentation

jimenezz22 reviewed (changes requested):
To explain the use of torii in the client please use this [example](https://github.com/AkatsukiLabs/Dojo-Game-Starter/blob/main/client/src/dojo/hooks/usePlayer.tsx) it's more meaningful than the example of x and y movements you are using

respp committed:
docs: update main Torii client integration guide to use Player model

- Replace Position/Moves examples with Player model (experience, health, coins, creation_day)
- Update GraphQL queries to use fullStarterReactPlayerModels
- Update component examples to display meaningful player statistics
- Update data conversion examples to handle player data types
- Improve prerequisites to reference Player model instead of Position/Moves

This makes the documentation more meaningful and aligns with real-world
Dojo Game Starter implementation patterns.

respp committed:
docs: update custom hooks guide to use Player model examples

- Replace Position/Moves hook examples with Player model hooks
- Update GraphQL queries to use fullStarterReactPlayerModels
- Update multiple model hook to use Player model structure
- Update enhanced hook with data conversion for player data
- Maintain consistent Player interface across all hook examples

This provides developers with practical, real-world hook patterns
that match the Dojo Game Starter implementation.

respp committed:
docs: update GraphQL queries guide to use Player model

- Replace Position/Moves query examples with Player model queries
- Update query structure to use fullStarterReactPlayerModels
- Update leaderboard query to sort by experience instead of remaining moves
- Update filtered query to use health conditions
- Update pagination query to use Player model fields
- Update multiple model query to use Player model structure

This provides developers with practical GraphQL query patterns
that match real-world Dojo game implementations.

respp committed:
docs: update data conversion guide to use Player model examples

- Replace Position data conversion examples with Player model examples
- Update hex conversion utilities to handle player data fields
- Update bulk data conversion examples to use player data structure
- Update type-safe conversion interfaces for Player model
- Update generic type converter examples for player data
- Maintain consistent Player interface across all conversion examples

This provides developers with practical data conversion patterns
for handling Cairo model data in real-world applications.

respp committed:
docs: update error handling guide to use Player model examples

- Replace Position error handling examples with Player model examples
- Update hook with error handling to use Player model structure
- Update component usage example to display player statistics
- Update fallback example to use player data instead of position data
- Update GraphQL queries to use fullStarterReactPlayerModels
- Maintain consistent error handling patterns for player data

This provides developers with robust error handling patterns
that work with real-world Player model implementations.

respp committed:
docs: update performance guide to use Player model examples

- Replace Position/Moves performance examples with Player model examples
- Update optimized hook to use Player model structure
- Update request batching examples to use Player model queries
- Update pagination query examples to use Player model fields
- Update usage examples to demonstrate Player model optimization
- Update GraphQL queries to use fullStarterReactPlayerModels

This provides developers with performance optimization patterns
that work with real-world Player model implementations.

respp committed:
docs: update setup guide to use Player model examples

- Replace Position/Moves prerequisites with Player model prerequisites
- Update debug query example to use fullStarterReactPlayerModels
- Update model verification examples to use Player model structure
- Maintain consistent Player model references throughout setup guide

This ensures the setup guide aligns with the Player model examples
used throughout the rest of the Torii integration documentation.
